fossunited/api/hackathon.py

```diff
@@ -565,3 +565,81 @@ def validate_user_as_localhost_member(localhost_id: str):
         )
 
     return True
+
+
+@frappe.whitelist()
+def add_pr_issue_to_project(project: str, details: dict) -> None:
+    """
+    Add a PR/Issue to a project
+
+    Args:
+        project (str): Project ID
+        details (dict): PR/Issue details
+    """
+    if not frappe.db.exists("FOSS Hackathon Project", project):
+        frappe.throw("Project does not exist")
+
+    issue_pr = frappe.get_doc(
+        {
+            "doctype": "Hackathon Project Issue PR",
+            "parent": project,
+            "parenttype": "FOSS Hackathon Project",
+            "parentfield": "issue_pr_table",
+            "title": details["title"],
+            "link": details["link"],
+            "type": details["type"],
+        }
+    )
+    issue_pr.insert()
+
+
+@frappe.whitelist()
+def remove_pr_issue_from_project(project: str, issue_pr: str) -> None:
+    """
+    Remove a PR/Issue from a project
+
+    Args:
+        project (str): Project ID
+        issue_pr (str): Issue/PR ID
+    """
+    if not frappe.db.exists("FOSS Hackathon Project", project):
+        frappe.throw("Project does not exist")
+
+    if not frappe.db.exists("Hackathon Project Issue PR", issue_pr):
+        frappe.throw("Issue/PR does not exist")
+
+    frappe.db.delete("Hackathon Project Issue PR", issue_pr)
+
+
+@frappe.whitelist()
+def get_issue_pr_title(url: str) -> dict:
+    """
+    Get title of the issue/PR/Discussion from the URL
+
+    Args:
+        url (str): URL of the issue/PR/Discussion
+
+    Returns:
+        dict: Issue/PR title
+    """
+    if not "https://github.com" in url:
```

Comment:
PR need not be raised only in Github - it could be Gitlab too. 

This condition would throw errors for gitlab cases. 

Now thinking of it, what would be the case for PRs made in Gitlab ? - Like the one for [Internet shutdowns](https://gitlab.com/sflcin/internetshutdowns.in/-/issues/?sort=updated_desc&state=opened&first_page_size=20) project.